io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly3em/IoShelly3EmImpl.java

```diff
@@ -160,34 +161,43 @@ private void processHttpResult(JsonElement result, Throwable error) {
 
 				var relays = getAsJsonArray(response, "relays");
 				if (!relays.isEmpty()) {
-					relay = getAsBoolean(getAsJsonObject(relays.get(0)), "ison");
+					var relay = getAsJsonObject(relays.get(0));
+					relay0 = getAsBoolean(relay, "ison");
+					overpower = getAsBoolean(relay, "overpower");
 				}
 
 				var update = getAsJsonObject(response, "update");
 				hasUpdate = getAsBoolean(update, "has_update");
 
+				var totalPower = getAsFloat(response, "total_power");
 				var emeters = getAsJsonArray(response, "emeters");
 				for (int i = 0; i < emeters.size(); i++) {
 					var emeter = getAsJsonObject(emeters.get(i));
 					var power = round(getAsFloat(emeter, "power"));
-					var voltage = round(getAsFloat(emeter, "voltage"));
-					var current = round(getAsFloat(emeter, "current"));
+					var voltage = round(getAsFloat(emeter, "voltage") * 1000);
+					var current = round(getAsFloat(emeter, "current") * 1000);
+					var isValid = getAsBoolean(emeter, "is_valid");
+
+					activePower = round(totalPower);
```

Comment:
This could be before the for-loop, right?

Just to `activePower = getAsFloat(response, "total_power");` in line 172